platform/dataprovider/src/main.rs

```diff
@@ -0,0 +1,76 @@
+use actix_web::{get, Responder, HttpRequest, HttpResponse, HttpServer, App};
+use actix_web::web::Query;
+use pocketsizefund::data::Client as DataClient;
+use pocketsizefund::data::client::Bar;
+use chrono::DateTime;
+use std::collections::HashMap;
+
+#[get("/health")]
+async fn health_handler() -> impl Responder {
+    HttpResponse::Ok().body("OK")
+}
+
+#[get("/data")]
+async fn data_handler(request: HttpRequest) -> impl Responder {
+    let request_url = request.full_url();
+
+    let query_string = request_url.query().unwrap();
+
+    let query_parameters = Query::<HashMap<String, String>>::from_query(query_string).unwrap();
+
+    let start_at = DateTime::parse_from_rfc3339(query_parameters.get("start_at").unwrap()).unwrap();
+    
+    let end_at = DateTime::parse_from_rfc3339(query_parameters.get("end_at").unwrap()).unwrap();
+
+    let data_client = DataClient::new(
+        std::env::var("ALPACA_API_KEY_ID").unwrap(),
+        std::env::var("ALPACA_API_SECRET_KEY").unwrap(),
+        std::env::var("AWS_ACCESS_KEY_ID").unwrap(),
+        std::env::var("AWS_SECRET_ACCESS_KEY").unwrap(),
+        std::env::var("S3_DATA_BUCKET_NAME").unwrap(),
+    );
+
+    let old_bars = data_client.load_equities_bars().await.unwrap();
+
+    let filtered_bars: Vec<Bar> = old_bars
+        .into_iter()
+        .filter(|bar| bar.timestamp >= start_at && bar.timestamp <= end_at)
+        .collect();
+
+    HttpResponse::Ok().json(filtered_bars)
+}
+
+#[actix_web::main]
+async fn main() -> std::io::Result<()> {
+    HttpServer::new(|| {
+        App::new()
+            .service(health_handler)
+            .service(data_handler)
+    })
+    .bind(("127.0.0.1", 3000))?
```

Comment:
Dockerfile is exposing `8080`, we could just make this an `env::var("PORT", 8080)` might be good to just have it passed in as a general rule